### PR TITLE
Added dependency injection container in legacy

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Allow us to build our objects a more flexible and powerful way
| Type?         | improvement
| Category?     | CO
| Test            | In legacy controllers, use $this->get('sf_finder') for instance

- [x] Todos:
  - [x] Bootstrap a container for both legacy admin and front office environments

### CODE REVIEW

File: src/Core/Stock/StockManager.php [hunk 1]
```diff
@@ -23,44 +23,65 @@
...
+use PrestaShop\PrestaShop\Adapter\StockManager as StockManagerAdapter;
+use PrestaShop\PrestaShop\Adapter\CacheManager;
+use PrestaShop\PrestaShop\Adapter\HookManager;
+
 /**
  * Class StockManager Refactored features about product stocks.
- *
- * @package PrestaShop\PrestaShop\Core\Stock
  */
 class StockManager
```
Comment (mickaelandrieu) [inline src/Core/Stock/StockManager.php]: Note this class are not used anymore, but will be in ``1.7.1``

File: src/Adapter/Product/PackItemsManager.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
```
Comment (mickaelandrieu) [inline src/Adapter/Product/PackItemsManager.php]: Note this class are not used anymore, but will be in ``1.7.1``.

**Comment by Quetzacoalt91** [comment]
seal_of_approval_1

**Comment by Shine-neko** [comment]
ping @Shudrum @thierrymarianne  We merge ? 😃

**Comment by Quetzacoalt91** [comment]
We could merge this one for 1.7.1.0 ?

**Comment by aleeks** [comment]
Let's go to merge this ? @mickaelandrieu @Quetzacoalt91

**Comment by mickaelandrieu** [comment]
it's not up to me to decide on my own pr :)

File: src/Adapter/ServiceLocator.php [hunk 3]
```diff
@@ -27,6 +27,11 @@
 
 use \PrestaShop\PrestaShop\Core\Foundation\IoC\Container;
 
+/**
+ * @internal
+ *
+ * To be removed in 1.7.1.
```
Comment (mickaelandrieu) [inline src/Adapter/ServiceLocator.php]: To be removed in 1.8 ? lol

File: config/services/admin/services_dev.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+imports:
+    - { resource: services_prod.yml }
```
Comment (mickaelandrieu) [inline config/services/admin/services_dev.yml]: @thierrymarianne can help you a lot with unit testing :)

**Review by thierrymarianne — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
PR rebased, issue on BO fixed :+1:

**Comment by aleeks** [comment]
Thank you @mickaelandrieu

**Pull request merged**